src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.GroupedAnalyzerActionsByAnalyzer.cs

```diff
@@ -0,0 +1,193 @@
+            private ImmutableArray<CodeBlockAnalyzerAction> _lazyCodeBlockActions;
+            private ImmutableArray<OperationBlockStartAnalyzerAction> _lazyOperationBlockStartActions;
+            private ImmutableArray<OperationBlockAnalyzerAction> _lazyOperationBlockActions;
+            private ImmutableArray<OperationBlockAnalyzerAction> _lazyOperationBlockEndActions;
+
+            public GroupedAnalyzerActionsForAnalyzer(DiagnosticAnalyzer analyzer, in AnalyzerActions analyzerActions, bool analyzerActionsNeedFiltering)
+            {
+                Debug.Assert(!analyzerActions.IsEmpty);
+
+                _analyzer = analyzer;
+                AnalyzerActions = analyzerActions;
+                _analyzerActionsNeedFiltering = analyzerActionsNeedFiltering;
+            }
+
+            public AnalyzerActions AnalyzerActions { get; }
+
+            [Conditional("DEBUG")]
+            private static void VerifyActions<TAnalyzerAction>(in IEnumerable<TAnalyzerAction> actions, DiagnosticAnalyzer analyzer)
+                where TAnalyzerAction : AnalyzerAction
+            {
+                foreach (var action in actions)
+                {
+                    Debug.Assert(action.Analyzer == analyzer);
+                }
+            }
+
+            private ImmutableArray<TAnalyzerAction> GetFilteredActions<TAnalyzerAction>(in ImmutableArray<TAnalyzerAction> actions)
+                where TAnalyzerAction : AnalyzerAction
+                => GetFilteredActions(actions, _analyzer, _analyzerActionsNeedFiltering);
+
+            private static ImmutableArray<TAnalyzerAction> GetFilteredActions<TAnalyzerAction>(
+                in ImmutableArray<TAnalyzerAction> actions,
+                DiagnosticAnalyzer analyzer,
+                bool analyzerActionsNeedFiltering)
+                where TAnalyzerAction : AnalyzerAction
+            {
+                if (!analyzerActionsNeedFiltering)
+                {
+                    return actions;
+                }
+
+                var builder = ArrayBuilder<TAnalyzerAction>.GetInstance();
+                foreach (var action in actions)
+                {
+                    if (action.Analyzer == analyzer)
+                    {
+                        builder.Add(action);
+                    }
+                }
+
+                return builder.ToImmutableAndFree();
+            }
+
+            public ImmutableDictionary<TLanguageKindEnum, ImmutableArray<SyntaxNodeAnalyzerAction<TLanguageKindEnum>>> NodeActionsByAnalyzerAndKind
+            {
+                get
+                {
+                    if (_lazyNodeActionsByKind == null)
+                    {
+                        var nodeActions = _analyzerActionsNeedFiltering ?
+                            AnalyzerActions.GetSyntaxNodeActions<TLanguageKindEnum>(_analyzer) :
+                            AnalyzerActions.GetSyntaxNodeActions<TLanguageKindEnum>();
```

Comment:
```suggestion
                        var nodeActions = _analyzerActionsNeedFiltering
                            ? AnalyzerActions.GetSyntaxNodeActions<TLanguageKindEnum>(_analyzer)
                            : AnalyzerActions.GetSyntaxNodeActions<TLanguageKindEnum>();
```

Reply:
applies to all.

Reply:
The current pattern matches the code style used throughout this layer.